domains/Accounts/Controllers/AccountsProfileController.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Domains\Accounts\Controllers;
+
+use App\Http\Controllers\Controller;
+use Domains\Accounts\Resources\UserResource;
+use Illuminate\Http\Response;
+use Illuminate\Contracts\Auth\Factory as Auth;
+
+class AccountsProfileController extends Controller
+{
+    protected $auth;
+
+    public function __construct(Auth $auth)
+    {
+        $this->auth = $auth;
+    }
+
+    public function __invoke(): Response
+    {
+        return new Response(UserResource::make($this->auth->user()), Response::HTTP_OK);
```

Comment:
We don't need to use the response. We can return the `UserResource`, directly, and the framework will return a standardized response structure. Of course, we need to also change the method's return type.

Reply:
Ok, I use this because by default the framework return http status 201

Reply:
@josepostiga, I use this because the framework return 201 http status code. Do I make the change?

Reply:
Yes. 

That happens because the test suite creates an instance of `User` in the same request lifecycle. The `User` model's `wasRecentlyCreated` property will be true, and since the framework looks up that property value to know which response status code to return (check `ResourceResponse`, `calculateStatus` method) it gets tricked to think that the resource was created in the same request. Technically, it was, but in reality, it won't.

On the test method we have an explicit check for the status code. which we can drop (unfortunately, Lumen doesn't have the `assertSuccessfully` method, which would be perfect for this). We can assume that since we can see the JSON response, everything was ok

Reply:
Thanks for the explanation. I will change.

Reply:
Thank you for your explanation.